src/actions/app.js

```diff
@@ -168,19 +168,31 @@ export const { hideAlert } = createActions({
 
 export const { runQuery } = createActions({
   [ActionTypes.RUN_QUERY_SUCCESS]: (result) => result,
-  [ActionTypes.RUN_QUERY]: (projectId, query, maxResults) => ({
+  [ActionTypes.RUN_QUERY]: (projectId, query) => ({
     projectId,
     query,
-    maxResults,
   }),
 });
 
+export const { previewData } = createActions({
+  [ActionTypes.PREVIEW_DATA]: (snapshotId, table, columns, totalRowCount) => ({
```

Comment:
Will we be using the preview API for datasets as well as snapshots? If so, we might want to not call the id `snapshotId` here.